pilot/pkg/xds/delta.go

```diff
@@ -318,12 +320,35 @@ func (s *DiscoveryServer) processDeltaRequest(req *discovery.DeltaDiscoveryReque
 	if con.proxy.SidecarScope != nil && con.proxy.SidecarScope.Version != request.Push.PushVersion {
 		s.computeProxyState(con.proxy, request)
 	}
-	return s.pushDeltaXds(con, con.proxy.GetWatchedResource(req.TypeUrl), request)
+
+	err := s.pushDeltaXds(con, con.proxy.GetWatchedResource(req.TypeUrl), request)
+	if err != nil {
+		return err
+	}
+
+	select {
+	case dwr := <-dependentResourceChan:
```

Comment:
I don't think we need this.

IMO we can just do

```
if req.TypeUrl == cluster && proxy.GetWatchedResource(endpoints) != nil {
 ...
}
```

Reply:
Ack - I thought we needed to move the watchedResource; good to know there's a func for it